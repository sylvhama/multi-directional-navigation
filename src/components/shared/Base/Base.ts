import styled from "styled-components";

import { IsFocusedProps } from "components/shared/types";

export const Base = styled.article<IsFocusedProps>`
  border: 5px solid #333;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 4rem;
  background-color: #dfe3e8;
  color: #212b36;

  ${({ isFocused }) =>
    isFocused &&
    `
      border-color: #006FBB;
      background-color: #F9FAFB;
      font-weight: bold;
    `}
`;
