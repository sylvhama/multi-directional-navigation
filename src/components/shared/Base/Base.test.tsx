import "jest-styled-components";
import React from "react";
import { render } from "@testing-library/react";

import { Base } from "./Base";

describe.each([true, false])("<Base />", (isFocused) => {
  it(`renders style when isFocused is ${isFocused}`, () => {
    const { container } = render(<Base isFocused={isFocused} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
