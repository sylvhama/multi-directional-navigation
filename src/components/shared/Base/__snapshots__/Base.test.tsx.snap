// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Base /> renders style when isFocused is false 1`] = `
.c0 {
  border: 5px solid #333;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 4rem;
  background-color: #dfe3e8;
  color: #212b36;
}

<article
  class="c0"
/>
`;

exports[`<Base /> renders style when isFocused is true 1`] = `
.c0 {
  border: 5px solid #333;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 4rem;
  background-color: #dfe3e8;
  color: #212b36;
  border-color: #006FBB;
  background-color: #F9FAFB;
  font-weight: bold;
}

<article
  class="c0"
/>
`;
